geomstats/geometry/fiber_bundle.py

```diff
@@ -489,3 +490,120 @@ def integrability_tensor_derivative(
             in Kendall shape spaces. Unpublished.
         """
         raise NotImplementedError
+
+
+class ProductFiberBundle(FiberBundle):
```

Comment:
I think "ProductFiberBundle" seems to be saying that we have a product: total_space_1 x total_space_2 since "fiber bundle" often refers to the underlying total_space in the math literature.

This could be confusing since what we want is actually one single total_space with two group actions, and not the product of two total_spaces each with its group action.

My question: why having a new class ProductFiberBundle at all? Can't we put this into FiberBundle directly?

The Rationale being that:

Class Names <-> only refer to the Mathematics: here, we have one single fiberbundle, with one single action, of a more complicated group (a product group) that we _implement_ as a product.

But "Product" should not appear in the class name, because it's purely an implementation choice (and an approximation performed in the alignment procedure). Or, make that class name private.

Reply:
Here `Product` comes from a product of groups, not product of total spaces. I can see why looking to the implementation may point towards the second.

Since what defines this fiber bundle is the alignment algorithm, to avoid creating this new bundle, I suggest implementing it as an `AlignerAlgorithm` (a bit like e.g. `DynamicProgramming`) and then use the same bundle. Would this be preferable to you? Probably is what has more logical sense.

Thanks for raising this point!

Reply:
@ninamiolane, I've addressed this comment, please let me know what you think. Implementation hasn't changed, it is really just a matter of code organization. I think this version makes more math/logical sense.